### PR TITLE
Fix revapi release branch

### PR DESCRIPTION
**Pull request by yunikkk**
### Summary of changes
Fixed revapi check both for release and main branches. Currently it 
- fails on PRs to release branch picking incorrect `android-v10.6.0` release - [the build](https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android/6793/workflows/aad95f19-6525-40e9-b090-44692f009e5f/jobs/57080)
```
$1:
$2:./plugin-compass/build/outputs/aar/plugin-compass-release.aar
$3:compass
$4:android-v10.6.0
Path to previous version of aar is not set, using 10.6.0
aar path is: s3://mapbox-api-downloads-production/v2/mobile-maps-android-compass/releases/android/10.6.0/maven/maps-compass-10.6.0.aar, checking file
```

- succeedes without any actual comparison on PRs to the main branch - [the build](https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android/6781/workflows/50301da3-815e-4f4a-9fa1-c33a39d85529/jobs/56967)
```
$1:
$2:./plugin-viewport/build/outputs/aar/plugin-viewport-release.aar
$3:viewport
$4:v10.8.0
Path to previous version of aar is not set, using 
aar path is: s3://mapbox-api-downloads-production/v2/mobile-maps-android-viewport/releases/android//maven/maps-viewport-.aar, checking file
s3://mapbox-api-downloads-production/v2/mobile-maps-android-viewport/releases/android//maven/maps-viewport-.aar doesn't exist on s3, skipping the check
```

After the change script correctly compares the previous release, e.g. [build](https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android/6796/workflows/041ddb7b-4828-4e14-b947-c5d1c23b3f56/jobs/57187) with `10.9.0-rc.1`
```
$1:
$2:./plugin-viewport/build/outputs/aar/plugin-viewport-release.aar
$3:viewport
$4:v10.9.0-rc.1
Path to previous version of aar is not set, using 10.9.0-rc.1
aar path is: s3://mapbox-api-downloads-production/v2/mobile-maps-android-viewport/releases/android/10.9.0-rc.1/maven/maps-viewport-10.9.0-rc.1.aar, checking file
Downloading file from s3

```

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [x] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: scripts/java-api-check-all.sh [hunk 1]
```diff
@@ -38,17 +38,17 @@ CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
...
 echo "TARGET_BRANCH: $TARGET_BRANCH"
-readonly RELEASE_TAG_PATTERN="v10\.[0-9]*\.[0-9]*$"
 readonly STABLE_RELEASE_TAG_PATTERN="v10\.[0-9]*\.0$"
 
 if ! [ "$TARGET_BRANCH" = "main" ]; then # Release branches
-  LATEST_RELEASE_TAG_IN_THIS_BRANCH=$(git tag --merged "$CURRENT_BRANCH" --sort=-creatordate | grep $RELEASE_TAG_PATTERN | head -n 1)
-  if [ "$LATEST_RELEASE_TAG_IN_THIS_BRANCH" = "*beta*" ] || [ "$LATEST_RELEASE_TAG_IN_THIS_BRANCH" = "*rc*" ]; then
-    # Use the latest stable minor release tag during rc and beta release
+  # filter tags present in the current release branch
+  LATEST_RELEASE_TAG_IN_THIS_BRANCH=$(git tag --sort=-creatordate | { grep "^$TARGET_BRANCH" || test $? = 1; } | head -n 1) # grep sets error if result is empty, so test is needed
```
Comment (yunikkk) [inline scripts/java-api-check-all.sh:45]: Last tag in the release branch, named with the release branch prefix is used to run comparison. 
e.g. for the current branch `v10.9` we take the last created `v10.9.0-rc.1`. 
If there's no tag (e.g. first beta release is created) - previous stable release will be used for comparison.

File: scripts/java-api-check.sh [hunk 2]
```diff
@@ -50,11 +55,11 @@ PREVIOUS_RELEASE_DIR=$(dirname "${PREVIOUS_RELEASE}")
 
 LAST_STABLE_VERSION_TAG=$4 #v10.3.0
 LAST_STABLE_VERSION_TAG_ARRAY=($LAST_STABLE_VERSION_TAG)
-LAST_STABLE_VERSION=${LAST_STABLE_VERSION_TAG_ARRAY[0]:9}
+LAST_STABLE_VERSION=${LAST_STABLE_VERSION_TAG_ARRAY[0]:1}
```
Comment (yunikkk) [inline scripts/java-api-check.sh:58]: Empty subscring was taken so on main branch job failed despite being green [here](https://app.circleci.com/pipelines/github/mapbox/mapbox-maps-android/6781/workflows/50301da3-815e-4f4a-9fa1-c33a39d85529/jobs/56967)

File: scripts/java-api-check-all.sh (hunk 1 above)
Comment (pengdev) [reply scripts/java-api-check-all.sh:45]: shouldn't we always compare to the last stable minor release?

Comment (yunikkk) [reply scripts/java-api-check-all.sh:45]: That's an open question. My thinking here was - if `10.9.0-beta.1` is compatible with the previous `10.8.5` - then `10.9.0-rc.1` will be as well. Then it should be enough to compare `10.9.0-rc.1` and `10.9.0-beta.1`. WDYT?

File: scripts/java-api-check-all.sh [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-set -Eeuo pipefail
+set -Exeuo pipefail
```
Comment (OdNairy) [inline scripts/java-api-check-all.sh:3]: Should it be a part of the public script? Looks more like a debug option for me

**Review by OdNairy — APPROVED** [review]

File: scripts/java-api-check-all.sh (hunk 3 above)
Comment (yunikkk) [reply scripts/java-api-check-all.sh:3]: It is indeed added to debug the logic - will remove it before merge

File: scripts/java-api-check-all.sh (hunk 1 above)
Comment (yunikkk) [reply scripts/java-api-check-all.sh]: Reworked to correctly find the previous minor release

**Review by pengdev — APPROVED** [review]

**Pull request merged**